### ISSUE TITLE
Plans Overhaul: Upgrade logic may be missing functionality from previous/original handler

### ISSUE THREAD

**Issue by chriskmnds** [Automattic/wp-calypso#63186]
### Details

The [original plan upgrade logic](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/plan-features/index.jsx#L561) seems to have been ported over to the new plans only partially [here](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/plans/main.jsx#L122). We recently discovered a bug with missing coupon support, which we fixed roughly in https://github.com/Automattic/wp-calypso/pull/63084. Probably wise to revisit and address the rest, if any other missing critical logic / whether working logic that needs inclusion or things that need proper deprecation. Other previous query strings / redirect params seem to not have been ported over.

### Related

https://github.com/Automattic/wp-calypso/pull/63084

**Comment by chriskmnds** [comment]
Bumping this to `Pri High` as there may be more breakage lurking here, but hopefully not 🤞 

CC @southp would be nice to have a project, milestone, and labels in Calypso to be tracking these issues outside of the Trello board really :-)

**Comment by dkd903** [comment]
Apologies for the delayed updates on this, got busy with Reboot training stuff. I found out that the following logic sections are currently not handles in the new flow:

1. Redirecting the user to Add domain flow (with flag set)
2. Check for whether or not the plan is available for the user
3. Check for atmic site conversion
4. Is redirect set

Of the above four, I have made changes for the first 2 and currently working on the next two. I will submit the PR later today.

cc: @southp  @chriskmnds
